src/BizHawk.Client.EmuHawk/tools/TAStudio/MarkerControl.cs

```diff
@@ -254,6 +265,37 @@ public void EditMarkerPopUp(TasMovieMarker marker, bool followCursor = false)
 			}
 		}
 
+		public void EditMarkerFramePopUp(TasMovieMarker marker, bool followCursor = false)
+		{
+			var markerFrame = marker.Frame;
+			var i = new InputPrompt
+			{
+				Text = $"Marker for frame {markerFrame}",
+				TextInputType = InputPrompt.InputType.Unsigned,
+				Message = "Enter a frame number",
+				InitialValue =
+					Markers.IsMarker(markerFrame)
+					? Markers.PreviousOrCurrent(markerFrame).Frame.ToString()
+					: "0"
+			};
+
+			if (followCursor)
+			{
+				var point = Cursor.Position;
+				point.Offset(i.Width / -2, i.Height / -2);
+				i.StartPosition = FormStartPosition.Manual;
+				i.Location = point;
+			}
+
+			if (this.ShowDialogWithTempMute(i) == DialogResult.OK && !Markers.IsMarker(int.Parse(i.PromptText)))
+			{
+				Markers.Move(marker.Frame, int.Parse(i.PromptText));

```

Comment:
Might make sense to only parse once and then reuse the value.

Reply:
also `.IsOk()` instead of `== DialogResult.OK`